src/components/application_manager/src/commands/mobile/on_vehicle_data_notification.cc

```diff
@@ -56,13 +56,24 @@ void OnVehicleDataNotification::Run() {
   std::vector<smart_objects::SmartObject> appSO;
 
   const VehicleData& vehicle_data = MessageHelper::vehicle_data();
+  const smart_objects::SmartObject incoming_msg_params = (*message_)[strings::msg_params];
   VehicleData::const_iterator it = vehicle_data.begin();
+  std::string key;
 
   for (; vehicle_data.end() != it; ++it) {
-    if (true == (*message_)[strings::msg_params].keyExists(it->first)) {
+    key = it->first;
```

Comment:
@KVGrygoriev 
const std::string key = it->first;
would be more compact

Reply:
@EKuliiev each time creates new variable, I don't think it's a good idea
Look above, it's a cycle